lib/configuration.dart

```diff
@@ -0,0 +1,92 @@
+import 'constants.dart';
+import 'tracking_options.dart';
+import 'default_tracking.dart';
+
+/// Configuration for Amplitude instance.
+/// 
+/// Before initializing Amplitude instance, create a Configuration instance
+/// with your desired configuration and pass it to the Amplitude instance.
+/// Note the Configuration is immutable (cannot be changed) after being passed to Amplitude.init()
+/// `optOut` can be changed later by calling `setOptOut()`.
+class Configuration {
+  String apiKey;
+  int flushQueueSize;
+  int flushIntervalMillis;
+  late String instanceName;
+  bool optOut;
+  LogLevel logLevel;
+  int? minIdLength;
+  String? partnerId;
+  int flushMaxRetries;
+  bool useBatch;
+  ServerZone serverZone;
+  String? serverUrl;
+  int minTimeBetweenSessionsMillis;
+  DefaultTrackingOptions defaultTracking;
+  TrackingOptions trackingOptions;
+  // Mobile (iOS and Android) specific
```

Comment:
Nit: not sure how these appear in "IntelliSense" but I think it would be best to add an individual comment per parameter to be explicit. Also [based on this](https://dart.dev/effective-dart/documentation#do-use--doc-comments-to-document-members-and-types), I think `///` is needed for these to appear in generated documentation

```
  /// `enableCoppaControl` (iOS & Android only)
  bool enableCoppaControl;
  /// `flushEventsOnClose` (iOS & Android only)
  bool flushEventsOnClose;
 ...
  /// `locationListening` (Android only)
  bool locationListening;
```

Reply:
Yes I am using `///` for class descriptions. Added descriptions for class properties too at https://github.com/amplitude/Amplitude-Flutter/pull/162/commits/7624e9953d0cafd00527c45608fe999e7aa431bb
<img width="555" alt="image" src="https://github.com/amplitude/Amplitude-Flutter/assets/31029607/e356b189-c070-4464-be24-5dae0066e20d">